src/main/resources/db/migration/V14__create_auth_provider_table.sql

```diff
@@ -1,11 +1,16 @@
+-- OAuth Grant Type Enum
+CREATE TYPE grant_type AS ENUM (
+    'CODE_AUTHORIZATION',
+    'CUSTOM',
+    'IMPLICIT'
+);
+
+-- OAuth Provider Table
 CREATE TABLE auth_provider (
-    auth_url        VARCHAR(512) NOT NULL,
-    client_id       VARCHAR(512) NOT NULL,
-    client_secret   VARCHAR(512) NOT NULL,
-    provider        VARCHAR(512) NOT NULL,
-    grant_type      VARCHAR(512) NOT NULL,
-    scopes          JSONB NOT NULL,
+    provider_name    VARCHAR(512) NOT NULL PRIMARY KEY, -- OAuth Provider Name (e.g. Google, Microsoft)
+    auth_url        VARCHAR(512) NOT NULL,              -- OAuth Authorization URL that frontend consumes to redirect user
+    client_id       VARCHAR(512) NOT NULL,              -- OAuth Client ID generated at OAuth Provider to include in request
+    client_secret   VARCHAR(512) NOT NULL,              -- OAuth Client Secret generated at OAuth Provider to obtain refresh token
```

Comment:
postgre has `comment on` feature which you can use to add comments
- https://www.postgresql.org/docs/current/sql-comment.html
- https://stackoverflow.com/a/32071293

Reply:
Done